### PR TITLE
branchSharelinkDelegate doc update

### PR DESCRIPTION
**Pull request by agrimn**
Added code snippets to show how BranchSharelinkDelegate is inherited and allows Branch link customization on the fly.

### CODE REVIEW

**Comment by BranchMacMini** [comment]
Result of [Integration 1](https://stlt.herokuapp.com/v1/xcs_deeplink/qabot.stage.branch.io/54a4eba29c2f74a2f6b1c16d60959b36/54a4eba29c2f74a2f6b1c16d60959e42)
---
*Duration*: 5 minutes and 18 seconds
*Result*: **Perfect build!** All 179 tests passed. :+1:
*Test Coverage*: 57%

File: README.md [hunk 1]
```diff
@@ -910,6 +910,44 @@ linkProperties.addControlParam("$email_html_footer", withValue: "Thanks!")
 linkProperties.addControlParam("$email_html_link_text", withValue: "Tap here")
 ```
 
+#### Changing share Text on the fly
+
+You have the option of changing the link shareText and other link parameters based on the choice the user makes on the Sharesheet Activity. For this to work your View Controller will need to inherit the BranchShareLink Delegate.
```
Comment (E-B-Smith) [inline README.md]: Can you change:

> You have the option of changing the link shareText and other link parameters based on the choice the user makes on the Sharesheet Activity. For this to work your View Controller will need to inherit the BranchShareLink Delegate.

to something like:

> You can change the link shareText and other link parameters based on the choice the user makes on the sharesheet activity.  First, set the `BranchShareLink` delegate with an object that follows the `BranchShareLinkDelegate` protocol.

> The optional `- (void) branchShareLinkWillShare:(BranchShareLink*)shareLink` delegate method will be called just after the user selects a share action, like share by email for instance, and before the share action is shown to the user, like when the email composer is shown to the user with the share text. This is an ideal time to change the share text based on the user action.

> The optional `- (void) branchShareLink:didComplete:withError:` delegate method will be called after the user has completed the share action.  The `didComplete` boolean will be `YES` if the user shared the item, and `NO` if the user cancelled.  The `error` value will indicate any errors that may have occurred.

**Comment by BranchMacMini** [comment]
Result of [Integration 2](https://stlt.herokuapp.com/v1/xcs_deeplink/qabot.stage.branch.io/54a4eba29c2f74a2f6b1c16d60959b36/54a4eba29c2f74a2f6b1c16d60fea11e)
---
*Duration*: 4 minutes and 45 seconds
*Result*: **Perfect build!** All 179 tests passed. :+1:
*Test Coverage*: 57%

**Comment by BranchMacMini** [comment]
Result of [Integration 3](https://stlt.herokuapp.com/v1/xcs_deeplink/qabot.stage.branch.io/54a4eba29c2f74a2f6b1c16d60959b36/a5a9885a62b6c6c7d54314342d1f7bec)
---
*Duration*: 4 minutes and 31 seconds
*Result*: **Perfect build!** All 179 tests passed. :+1:
*Test Coverage*: 57%

**Pull request merged**